extensions/sql/contract-definition/loader/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/LazyDataSource.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (c) 2022 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.contractdefinition.loader;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+
+class LazyDataSource implements DataSource {
```

Comment:
Is there any particular reason why you would want the DataSource lazily initialized? Also, since it actually _wraps_ a datasource, it should be named `LazyDataSourceWrapper` to make this visible.